planning/behavior_path_planner/src/scene_module/start_planner/start_planner_module.cpp

```diff
@@ -492,99 +492,109 @@ void StartPlannerModule::planWithPriority(
   const std::vector<Pose> & start_pose_candidates, const Pose & refined_start_pose,
   const Pose & goal_pose, const std::string search_priority)
 {
-  // check if start pose candidates are valid
-  if (start_pose_candidates.empty()) {
-    return;
-  }
-
-  const auto is_safe_with_pose_planner = [&](const size_t i, const auto & planner) {
-    // Get the pull_out_start_pose for the current index
-    const auto & pull_out_start_pose = start_pose_candidates.at(i);
-
-    // Set back_finished to true if the current start pose is same to refined_start_pose
-    status_.driving_forward =
-      tier4_autoware_utils::calcDistance2d(pull_out_start_pose, refined_start_pose) < 0.01;
-
-    planner->setPlannerData(planner_data_);
-    const auto pull_out_path = planner->plan(pull_out_start_pose, goal_pose);
-    // not found safe path
-    if (!pull_out_path) {
-      return false;
-    }
-    // use current path if back is not needed
-    if (status_.driving_forward) {
-      const std::lock_guard<std::mutex> lock(mutex_);
-      status_.found_pull_out_path = true;
-      status_.pull_out_path = *pull_out_path;
-      status_.pull_out_start_pose = pull_out_start_pose;
-      status_.planner_type = planner->getPlannerType();
-      return true;
-    }
+  if (start_pose_candidates.empty()) return;
 
-    // If this is the last start pose candidate, return false
-    if (i == start_pose_candidates.size() - 1) return false;
+  PriorityOrder order_priority =
+    determinePriorityOrder(search_priority, start_pose_candidates.size());
 
-    // check next path if back is needed
-    const auto & pull_out_start_pose_next = start_pose_candidates.at(i + 1);
-    const auto pull_out_path_next = planner->plan(pull_out_start_pose_next, goal_pose);
-    // not found safe path
-    if (!pull_out_path_next) {
-      return false;
-    }
+  for (const auto & pair : order_priority) {
+    if (findPullOutPath(
+          start_pose_candidates, pair.first, pair.second, refined_start_pose, goal_pose))
+      return;
+  }
```

Comment:
I prefer
```suggestion
  for (const auto & [index, planner] : order_priority) {
    if (findPullOutPath(
          start_pose_candidates, index, planner, refined_start_pose, goal_pose))
      return;
  }
```

Reply:
I didn't know this way and looks better!
changed in https://github.com/autowarefoundation/autoware.universe/pull/5393/commits/791145e6f8368693f6ed74cfa19b6fb33a7a7434